### ISSUE TITLE
🐛 BUG: Fatal compiler error when using any version above 0.21.6

### ISSUE THREAD

**Issue by edazpotato** [withastro/astro#2176]
I get the following error (with the `--verbose` flag) hen trying to do a build on any version of Astro above 0.21.6.
```
panic: runtime error: nil pointer dereference
RuntimeError: Error: Uh oh, the Astro compiler encountered an unrecoverable error!
```
That's all the logging there is. Sorry if this isn't too helpful yet, but I figured I should open an issue.
Is there a way to make the compiler give more information in the case that it crashes?

**Comment by jonathantneal** [comment]
I’m unable to replicate this in Astro v0.21.12, but I also have to put the flag after the command, like so:

```shell
astro dev --verbose
```

The documentation may need to be updated.

**Comment by edazpotato** [comment]
Yeah, v0.21.12 seems to have fixed it - I still get the error in v0.21.11.
